### PR TITLE
Save window size

### PR DESCRIPTION
**Pull request by thewizzy**
Save the window size on exit.

Should we remove the logic to save the slider until window closed too?

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -13,6 +13,7 @@ use crate::{
 };
 
 pub const DEFAULT_SIDEBAR_WIDTH: f64 = 140.0;
+pub const DEFAULT_WINDOW_SIZE: (f64, f64) = (800.0, 350.0);
```
Comment (dominikwilkowski) [inline src/config.rs:16]: At this point it might be nicer to just implement `Default` for `WindowSettings`?

Comment (thewizzy) [reply src/config.rs:16]: Good idea.

File: src/config.rs [hunk 2]
```diff
@@ -92,16 +92,23 @@ pub struct ConfigGeneral {
 #[derive(Clone, Debug, Serialize, Deserialize)]
 pub struct WindowSettings {
 	pub sidebar_width: f64,
+	pub window_size: (f64, f64),
+}
+impl Default for WindowSettings {
```
Comment (dominikwilkowski) [inline src/config.rs:97]:
```suggestion
}

impl Default for WindowSettings {
```
❤️

File: src/config.rs [hunk 3]
```diff
@@ -92,16 +92,24 @@ pub struct ConfigGeneral {
 #[derive(Clone, Debug, Serialize, Deserialize)]
 pub struct WindowSettings {
 	pub sidebar_width: f64,
+	pub window_size: (f64, f64),
+}
+
+impl Default for WindowSettings {
+	fn default() -> Self {
+		WindowSettings {
+			sidebar_width: DEFAULT_SIDEBAR_WIDTH,
```
Comment (dominikwilkowski) [inline src/config.rs:101]: I'd do the same here as well and use the `WindowSettings::default` for `DEFAULT_SIDEBAR_WIDTH` no?

Comment (thewizzy) [reply src/config.rs:101]: This is used for double click snap back to default.

Comment (dominikwilkowski) [reply src/config.rs:101]: Which you could just use the default as well no?

Comment (thewizzy) [reply src/config.rs:101]: I guess so. Just feels odd to create a window settings::default().sidebar_width

Comment (dominikwilkowski) [reply src/config.rs:101]: It's more semantic? Says what it does?

File: src/main.rs [hunk 4]
```diff
@@ -146,6 +146,14 @@ fn main() {
 								.entry(MenuItem::new("Menu item with something on the\tright"))
 							// menus are currently commented out in the floem codebase
 						})
+						.on_resize(move |r| {
+							window_size_config.general.write().window_settings.window_size =
+								(r.x1, r.y1);
+						})
+						.on_event(EventListener::WindowClosed, move |_| {
```
Comment (thewizzy) [inline src/main.rs:153]: This doesn't capture cmd+q :( 
It works for alt+f4 though. Need to look into it more.

Comment (thewizzy) [reply src/main.rs:159]: Implemented a debounce primitive. Closing this, it will now save 100ms after the last value sent.

**Review by dominikwilkowski — APPROVED** [review]

**Comment by thewizzy** [comment]
The latest commit changes it to using a call counter of type u64 which is guarded by arc/rwlock to allow it to be cloned and put into a bunch of closures without issues. 
I tried implementing with an Atomicu64 but there were issues particularly when trying to get the value access inside the closure exec_after. Can wrap it in an arc outside of this, maybe need to wrap it inside too.

**Comment by thewizzy** [comment]
> The latest commit changes it to using a call counter of type u64 which is guarded by arc/rwlock to allow it to be cloned and put into a bunch of closures without issues. I tried implementing with an Atomicu64 but there were issues particularly when trying to get the value access inside the closure exec_after. Can wrap it in an arc outside of this, maybe need to wrap it inside too.

Writing this comment made me think of a way we might be able to do it. [1201d74](https://github.com/dominikwilkowski/vault/pull/20/commits/1201d742d98a16fca518b3b63804d50639e46412) is an AtomicU64 version.

File: src/main.rs [hunk 5]
```diff
@@ -11,7 +11,7 @@ use floem::{
 	view::View,
 	views::{container, dyn_container, Decorators},
 	window::WindowConfig,
-	Application,
+	Application, EventPropagation,
```
Comment (dominikwilkowski) [inline src/main.rs:14]:
```suggestion
	Application,
	EventPropagation,
```

**Review by dominikwilkowski — APPROVED** [review]

**Pull request merged**